src/containers/NFT/NFTTabs/Offers.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useInfiniteQuery } from 'react-query'
+import { Link } from 'react-router-dom'
+import Loader from '../../shared/components/Loader'
+import './styles.scss'
+import noInfo from '../../shared/images/no_info.svg'
+import SocketContext from '../../shared/SocketContext'
+import { Amount } from '../../shared/components/Amount'
+import { analytics, ANALYTIC_TYPES } from '../../shared/utils'
+import '../../shared/components/TransactionTable/styles.scss' // Reuse load-more-btn
+
+const formatAmount = require('../../../rippled/lib/txSummary/formatAmount')
+
+interface Props {
+  tokenId: string
+  offerType: string
+  fetchOffers: (
+    socket: any,
+    id: string,
+    limit: number | undefined,
```

Comment:
you should write this as `limit?: number`

Reply:
I cant do that. I pass in either `getBuyNFToffers` or `getBuyNFToffers` into `fetchOffers`. Both of these two functions are in a javascript file, and js files dont support optional param for function

Reply:
They should support default values

Reply:
Yes  we need to have undefined type in order to use default param

```
const getBuyNFToffers = (rippledSocket, tokenId, limit = 50, marker = '') =>
  getNFToffers('nft_buy_offers', rippledSocket, tokenId, limit, marker)

```